patchwork-extensions-block/src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -1,1070 +1,1112 @@
+	 * @param pos   Block position in world
+	 * @return True to allow the entity to destroy this block
+	 */
+	default boolean canEntityDestroy(BlockState state, BlockView world, BlockPos pos, Entity entity) {
+		if (entity instanceof EnderDragonEntity) {
+			return !BlockTags.DRAGON_IMMUNE.contains(this.getBlock());
+		} else if ((entity instanceof WitherEntity) || (entity instanceof WitherSkullEntity)) {
+			return ((IForgeBlockState) state).isAir(world, pos) || WitherEntity.canDestroy(state);
+		}
+
+		return true;
+	}
+
+	// No call locations.
+	/**
+	 * Ray traces through the blocks collision from start vector to end vector returning a ray trace hit.
+	 *
+	 * @param state    The current state
+	 * @param world    The current world
+	 * @param pos      Block position in world
+	 * @param start    The start vector
+	 * @param end      The end vector
+	 * @param original The original result from {@link Block#collisionRayTrace(IBlockState, World, BlockPos, Vec3d, Vec3d)}
+	 * @return A result that suits your block
+	 */
+	@Nullable
+	default HitResult getRayTraceResult(BlockState state, World world, BlockPos pos, Vec3d start, Vec3d end, HitResult original) {
+		return original;
+	}
+
+	// TODO Call locations: Patches: Explosion*
+	/**
+	 * Determines if this block should drop loot when exploded.
+	 */
+	default boolean canDropFromExplosion(BlockState state, BlockView world, BlockPos pos, Explosion explosion) {
+		return state.getBlock().shouldDropItemsOnExplosion(explosion);
+	}
+
+	// Call locations: Patches: DebugHud (no todo because this works fine without it -- this method should never be overridden anyway.)
+	/**
+	 * Retrieves a list of tags names this is known to be associated with.
+	 * This should be used in favor of {@link net.minecraft.tag.TagContainer#getTagsFor}, as this caches the result.
+	 */
+	Set<Identifier> getTags();
+
+	// TODO Call locations: Patches: Explosion*
+	/**
+	 * Called when the block is destroyed by an explosion.
+	 * Useful for allowing the block to take into account block entities,
+	 * state, etc. when exploded, before it is removed.
+	 *
+	 * @param world     The current world
+	 * @param pos       Block position in world
+	 * @param explosion The explosion instance affecting the block
+	 */
+	default void onBlockExploded(BlockState state, World world, BlockPos pos, Explosion explosion) {
+		world.setBlockState(pos, Blocks.AIR.getDefaultState(), 3);
+		getBlock().onDestroyedByExplosion(world, pos, explosion);
+	}
+
+	// TODO Call locations: Patches: Entity*, ServerPlayerEntity*
+	/**
+	 * Determines if this block's collision box should be treated as though it can extend above its block space.
+	 * Use this to replicate fence and wall behavior.
+	 */
+	default boolean collisionExtendsVertically(BlockState state, BlockView world, BlockPos pos, Entity collidingEntity) {
+		return getBlock().matches(BlockTags.FENCES) || getBlock().matches(BlockTags.WALLS) || getBlock() instanceof FenceGateBlock;
+	}
+}
```

Comment:
I'll see if I can fix the commit for the \r\n here, potential force push incoming :/

Reply:
Let me know when the force-push is done and I'll review